File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        count = [0] * (201)
        result = 0
        for num in nums:
            if num >= 2 * diff:
                if count[num - diff] and count[num - 2 * diff]:
                    result += 1
            count[num] = 1
        return result

File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.java
class Solution {
    public int arithmeticTriplets(int[] nums, int diff) {
        int[] count = new int[201];
        int result = 0;
        for(int num : nums) {
            if(num >= 2 * diff)
                if(count[num - diff] == 1 && count[num - 2 * diff] == 1)
                    ++result;
            count[num] = 1;
        }
        return result;
    }
}